features/step_definitions/metadata.js

```diff
@@ -33,31 +33,24 @@ module.exports = function server() {
             throw new Error('jobName is neither BAR or BAZ');
         }
 
-        request({
+        return request({
             uri: `${this.instance}/${this.namespace}/jobs/${jobId}/builds`,
             method: 'GET',
             json: true
         }).then((response) => {
             this.buildId = response.body[0].id;
-            this.meta = response.body.meta;
 
             Assert.equal(response.statusCode, 200);
         });
     });
 
-    this.Then(/^add the { "foo": (.*) } to metadata in the "main" build container$/, () => null);
+    this.Then(/^add the { "foo": (?:.*) } to metadata in the "main" build container$/, () => null);
```

Comment:
I think what I would expect from a step definition like this would be that this would set up expected values you would check later.

For example:
```
this.Then(/^add the { "(.*)": "(.*)" } to metadata in the "main" build container$/, (key, value) => {
   this.expectedMeta = this.expectedMeta || {};
   this.expectedMeta[key] = value;
});
```

Then in `a record of the metadata is stored`:
```
Object.keys(this.expectedMeta).forEach((key) => {
  assert.equal(this.meta[key], this.expectedMeta[key]);
});
```